addons/stock/res_config.py

```diff
@@ -56,13 +53,9 @@ def set_group_locations(self, cr, uid, ids, context=None):
             The inverse logic about group_stock_multiple_locations is managed by the
             normal behavior of 'group_stock_multiple_locations' field.
         """
-        def ref(xml_id):
-            mod, xml = xml_id.split('.', 1)
-            return self.pool['ir.model.data'].get_object(cr, uid, mod, xml, context)
-
-        for obj in self.browse(cr, uid, ids, context=context):
-            config_group = ref('stock.group_single_location')
-            base_group = ref('base.group_user')
+        for obj in self:
+            config_group = self.env.ref('stock.group_single_location')
+            base_group = self.env.ref('base.group_user')
```

Comment:
In no need recordset  use only **id** of this reference variable then attach **.id** like : 
```
config_group_id = self.env.ref('stock.group_single_location').id
base_group_id = self.env.ref('base.group_user').id
```